lib/src/places.dart

```diff
@@ -483,6 +502,53 @@ class OpeningHours {
       json != null ? new OpeningHours(json["open_now"]) : null;
 }
 
+class OpeningHoursDetail extends OpeningHours {
+  final List<OpeningHoursPeriod> periods;
+  final List<String> weekdayText;
+
+  OpeningHoursDetail(openNow, this.periods, this.weekdayText) : super(openNow);
+
+  factory OpeningHoursDetail.fromJson(Map<String, dynamic> json) => json != null
+    ? new OpeningHoursDetail(
+        json["open_now"],
+        json["periods"]
+          ?.map((p) => new OpeningHoursPeriod.fromJson(p))
+          ?.toList()
+          ?.cast<OpeningHoursPeriod>(),
+        (json["weekday_text"] as List)?.cast<String>()
+      )
+    : null;
+}
+
+class OpeningHoursPeriodDate extends GoogleDateTime {
+  final int day;
+  final String time;
+
+  /// UTC Time
+  DateTime dateTime;
+
+  OpeningHoursPeriodDate(this.day, this.time) {
```

Comment:
@lejard-h I'm still fumbling my way through Dart, so please let me know if this constructor is unideal. It seemed the most efficient way to write this.